src/katello/agent/goferd/plugin.py

```diff
@@ -15,27 +15,35 @@
 The katello virtual agent.
 Provides content management APIs for pulp within the RHSM environment.
 """
+from six import PY2
+
 
 import os
 import sys
-import httplib
+
+if PY2:
+    import httplib as http
+else:
+    import http.client as http
 
 sys.path.append('/usr/share/rhsm')
-sys.path.append('/usr/lib/yum-plugins')
 
 from time import sleep
 from logging import getLogger
 from subprocess import Popen
 
```

Comment:
I was getting a stack trace from `path_monitor.add(REPOSITORY_PATH, send_enabled_report)` which I commented out locally to test. It looks like something will need to be updated for PY3 but not sure just what needs to happen there:

```
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [INFO][MainThread] gofer.agent.plugin:686 - plugin:katello loaded using: katello.agent.goferd.plugin
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 - plugin:katello, import failed
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 - Traceback (most recent call last):
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -   File "/usr/lib/python3.6/site-packages/gofer/agent/plugin.py", line 694, in _load
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -     plugin.load()
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -   File "/usr/lib/python3.6/site-packages/gofer/common.py", line 312, in sfn
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -     return fn(*args, **kwargs)
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -   File "/usr/lib/python3.6/site-packages/gofer/agent/plugin.py", line 440, in load
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -     self.delegate.loaded()
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -   File "/usr/lib/python3.6/site-packages/gofer/agent/decorator.py", line 89, in loaded
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -     fn()
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -   File "/usr/lib/python3.6/site-packages/katello/agent/goferd/plugin.py", line 86, in init_plugin
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -     path_monitor.add(path, certificate_changed)
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -   File "/usr/lib/python3.6/site-packages/gofer/common.py", line 312, in sfn
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -     return fn(*args, **kwargs)
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -   File "/usr/lib/python3.6/site-packages/gofer/pmon.py", line 167, in add
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -     self._paths.add(Tracker(path, target))
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -   File "/usr/lib/python3.6/site-packages/gofer/pmon.py", line 108, in __init__
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -     self.digest = digest(path)
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -   File "/usr/lib/python3.6/site-packages/gofer/pmon.py", line 68, in digest
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 -     _hash.update(s)
Apr 06 18:45:45 fedora27.strangeways.example.com goferd[13547]: [ERROR][MainThread] gofer.agent.plugin:697 - TypeError: Unicode-objects must be encoded before hashing
```

Reply:
Fixed and bulit in copr.  Because the version is based on the commit hash, you may need to reinstall.